### PR TITLE
Add Python TFRecord pipelines for read/write

### PR DESCRIPTION
**Pull request by Amar3tto**
Pipelines added:

1. **Read from TFRecord**
IO: TFRecord IO
SDK Language: Python
Run instruction:
- Create set of .tfrecord files
- run `python read_tfrecord.py --runner="DataflowRunner" --project="your-project" --region="your-region" --file_pattern="your-file-pattern"`

2. **Write to TFRecord**
IO: TFRecord IO
SDK Language: Python
Run instruction:
- Create folder for .tfrecord files
- run `python write_tfrecord.py --runner="DataflowRunner" --project="your-project" --region="your-region" --file_path_prefix="your-file-path-prefix"`

### CODE REVIEW

**Review by olehborysevych — APPROVED** [review]
LGTM
Please update PR description and move from draft

**Review by damondouglas — CHANGES_REQUESTED** [review]
Could we get a README in Python/tfrecord that describes how to execute and where people can go for more information?

File: Python/tfrecord/read_tfrecord.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import apache_beam as beam
+
+from apache_beam import Map
+from apache_beam.io.tfrecordio import ReadFromTFRecord
+from apache_beam.options.pipeline_options import PipelineOptions
+
+
+class TFRecordOptions(PipelineOptions):
+    @classmethod
+    def _add_argparse_args(cls, parser):
```
Comment (damccorm) [inline Python/tfrecord/read_tfrecord.py:25]: Can we add a comment similar to https://github.com/GoogleCloudPlatform/dataflow-cookbook/blob/549ad0308a907e72b8af7bd83371073eeb74eaf5/Python/bigquery/read_query_bigquery.py#L28 for these argparse args?

File: Python/tfrecord/read_tfrecord.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    with beam.Pipeline(options=options) as p:
+
+        output = (p | "Read from TFRecord" >> ReadFromTFRecord(
+                        file_pattern=options.file_pattern
+                    )
+                    | "Map from bytes" >> Map(map_from_bytes)
+                    | "Log Data" >> Map(logging.info))
+
+
+def map_from_bytes(element):
```
Comment (damccorm) [inline Python/tfrecord/read_tfrecord.py:49]: A comment explaining why we need this would be good

File: Python/tfrecord/write_tfrecord.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+from apache_beam import Create
+from apache_beam import Map
+from apache_beam.io.tfrecordio import WriteToTFRecord
+from apache_beam.options.pipeline_options import PipelineOptions
+
+
+class TFRecordOptions(PipelineOptions):
+    @classmethod
+    def _add_argparse_args(cls, parser):
+        parser.add_argument(
```
Comment (damccorm) [inline Python/tfrecord/write_tfrecord.py:27]: Same comment as above

File: Python/tfrecord/write_tfrecord.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+        output = (p | 'Create' >> Create(elements)
+                    | 'Map to Bytes' >> Map(map_to_bytes)
+                    | "Write to TFRecord" >> WriteToTFRecord(
+                        file_path_prefix=options.file_path_prefix
+                    ))
+
+
+def map_to_bytes(element):
+    import pickle
```
Comment (damccorm) [inline Python/tfrecord/write_tfrecord.py:60]: Same comment as above

File: Python/tfrecord/write_tfrecord.py (hunk 3 above)
Comment (damccorm) [reply Python/tfrecord/write_tfrecord.py:27]: This probably applies to most PRs

**Review by damccorm — COMMENTED** [review]
LGTM other than the comments

File: Python/tfrecord/read_tfrecord.py (hunk 2 above)
Comment (Amar3tto) [reply Python/tfrecord/read_tfrecord.py:56]: Done

File: Python/tfrecord/write_tfrecord.py (hunk 3 above)
Comment (Amar3tto) [reply Python/tfrecord/write_tfrecord.py:30]: Done

File: Python/tfrecord/write_tfrecord.py (hunk 4 above)
Comment (Amar3tto) [reply Python/tfrecord/write_tfrecord.py:72]: Done

**Review by damccorm — APPROVED** [review]
Thanks!

**Pull request merged**